src/neural-net-visualizer/visualizer/js/visualizer.js

```diff
@@ -1,3 +1,9 @@
+import {format} from 'util'
+console.log = function(...args) {
+  const message = format(...args);
+  mqttClient.publish('app/logs/neural-net-visualizer', message);
+  originalLog.apply(console, args);
+};
```

Comment:
```suggestion
const originalLog = console.log;
console.log = function(...args) {
  const message = `${args}`;
  if (mqtt_connected) {
    client.publish('app/logs/neural-net-visualizer', message);
  }
  originalLog.apply(console, args);
};
```

Reply:
```js
var client = null;
var mqtt_connected = false
```
This needs to be defined line ~278, and `mqtt_connected` needs to be set true/false in `onConnect` and `onDisconnect`